pagopa/pn-frontend
# Feature/PN-10855 - Sender Dashboard components 1/2

## Short description
Includes the development and integration of the first 3 statistics components of the sender dashboard

## List of changes proposed in this pull request
- add formatDateSMonth and oneYearAgo helpers to common date.utility
- add it translations for the short textual version of every month
- add it translations into statistics.json lang file
- change statistics.mock.ts to allow it choosing randomly between 4 different mocked data
- change PnECharts component's style to grant statistics components on the same row will have the same height
- add TrendStackedStatistics draft component to show data using multiple stacked lines
- add AggregateStatistics draft component to show aggregate statistics using pie chart
- add FiledNotificationsStatistics draft component to show accepted/refused statistics data
- add DeliveryModeStatistics draft component to show info about the number of notifications sent through digital and analogic channels
- add LastStateStatistics draft component to show statistics about the last state reached by notifications
- add AggregateAndTrendStatistics draft component to manage switching between aggregate and trend statistics visualization
- change Statistics.page adding 'last update' text
- incorporate FiledNotifications, LastState and DeliveryMode statistics component into Statistics.page

## How to test
- add a "IS_STATISTICS_ENABLED" env variable to your pa config.json and set it to true
- Start pn-pa-webapp and verify the statistics page loads properly and the displayed graphs change switching between "aggregate" and "trend" mode and moving from "days" to "weeks" timeframes.
